### PR TITLE
Prevent dead players from turning bar stools

### PR DESCRIPTION
**Pull request by Nopey**
## About the PR
Bugfix.

## Why / Balance
Previously, players could always turn a bar stool (or office chair, etc..) they were buckled into; even while stone cold dead.

## Technical details
RotateToFaceSystem now only lets players rotate barstools (& similar) when the actionBlockerSystem reports that they CanChangeDirection; this also means players who are frozen by admins (see AdminFrozenSystem) can no longer spin barstools.

SharedBuckleSystem no longer tries to prevent buckled players from being able to change direction, instead relying on RotateToFaceSystem to ensure that players on beds and such don't rotate.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame

Video showcasing prior broken behavior: 

https://github.com/space-wizards/space-station-14/assets/4543739/74be203a-0961-4850-842a-768d927f6691

## Breaking changes

**Changelog**
:cl:
- fix: Dead players can no longer spin on a bar stool.

### CODE REVIEW

**Comment by UbaserB** [comment]
This PR also resolves #22198

File: Content.Shared/Interaction/RotateToFaceSystem.cs [hunk 1]
```diff
@@ -80,14 +80,8 @@ public bool TryFaceCoordinates(EntityUid user, Vector2 coordinates, TransformCom
...
-            if (_actionBlockerSystem.CanChangeDirection(user))
-            {
-                if (!Resolve(user, ref xform))
-                    return false;
-
-                _transform.SetWorldRotation(xform, diffAngle);
-                return true;
-            }
+            if (!_actionBlockerSystem.CanChangeDirection(user))
+                return false;
```
Comment (metalgearsloth) [inline Content.Shared/Interaction/RotateToFaceSystem.cs:84]: Doesn't this break turrets or strafing?

Comment (Nopey) [reply Content.Shared/Interaction/RotateToFaceSystem.cs:84]: No, it doesn't (I tested walking around while in harm, and hostile turrets sure do still tear my poor character to shreds)

This change, combined with the change in SharedBuckleSystem, maintain the prior code's behavior except where an entity that Can(not)ChangeDirection is buckled onto another. Neither turrets nor strafing involve buckling.

**Comment by Nopey** [comment]
Rebased on latest master, no changes otherwise.

(Previous CI pipeline failure was flake).

**Pull request merged**